File: www/bar-stacked/demo-grid-light.m
//
//  ViewController.m
//  Highcharts
//
//  License: www.highcharts.com/license
//  Copyright © 2016 Highsoft AS. All rights reserved.
//

#import <UIKit/UIKit.h>
#import <Highcharts/Highcharts.h>

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    HIGChartView *chartView = [[HIGChartView alloc] initWithFrame:self.view.bounds];

    chartView.theme = @"grid-light";
    
    chartView.options = @{
    @"chart": @{
        @"type": @"bar"
    },
    @"title": @{
        @"text": @"Stacked bar chart"
    },
    @"xAxis": @{
        @"categories": @[
            @"Apples",
            @"Oranges",
            @"Pears",
            @"Grapes",
            @"Bananas"
        ]
    },
    @"yAxis": @{
        @"min": @0,
        @"title": @{
            @"text": @"Total fruit consumption"
        }
    },
    @"legend": @{
        @"reversed": @true
    },
    @"plotOptions": @{
        @"series": @{
            @"stacking": @"normal"
        }
    },
    @"series": @[
        @{
            @"name": @"John",
            @"data": @[
                @5,
                @3,
                @4,
                @7,
                @2
            ]
        },
        @{
            @"name": @"Jane",
            @"data": @[
                @2,
                @2,
                @3,
                @2,
                @1
            ]
        },
        @{
            @"name": @"Joe",
            @"data": @[
                @3,
                @4,
                @4,
                @2,
                @5
            ]
        }
    ]
};
    
    [self.view addSubview:chartView];
}

@end

File: www-swift/bar-stacked/demo-grid-light.swift
import Highcharts
import UIKit

class ViewController: UIViewController {

  override func viewDidLoad() {
    super.viewDidLoad()

    let chartView = HIChartView(frame: view.bounds)
    chartView.theme = "grid-light"

    let options = HIOptions()

    let chart = HIChart()
    chart.type = "bar"
    options.chart = chart

    let title = HITitle()
    title.text = "Stacked bar chart"
    options.title = title

    let xAxis = HIXAxis()
    xAxis.categories = ["Apples", "Oranges", "Pears", "Grapes", "Bananas"]
    options.xAxis = [xAxis]

    let yAxis = HIYAxis()
    yAxis.min = 0
    yAxis.title = HITitle()
    yAxis.title.text = "Total fruit consumption"
    options.yAxis = [yAxis]

    let legend = HILegend()
    legend.reversed = true
    options.legend = legend

    let plotOptions = HIPlotOptions()
    plotOptions.series = HISeries()
    plotOptions.series.stacking = "normal"
    options.plotOptions = plotOptions

    let credits = HICredits()
    credits.enabled = false
    options.credits = credits

    let john = HIBar()
    john.name = "John"
    john.data = [5, 3, 4, 7, 2]

    let jane = HIBar()
    jane.name = "Jane"
    jane.data = [2, 2, 3, 2, 1]

    let joe = HIBar()
    joe.name = "Joe"
    joe.data = [3, 4, 4, 2, 5]

    options.series = [john, jane, joe]

    chartView.options = options

    self.view.addSubview(chartView)
  }

}
